### PR TITLE
@wdio/junit-reporter - replace ansi chars in test.error.message

### PR DESCRIPTION
**Pull request by unickq**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

https://github.com/webdriverio/webdriverio/pull/6662 for JUnit - remove ANSI chars from **message** generated by Expect lib.

image

```xml
<?xml version="1.0" encoding="UTF-8"?>
<testsuites>
  <testsuite name="A" timestamp="2021-06-24T16:16:57" time="0.217" tests="1" failures="1" errors="1" skipped="0">
    <testcase classname="wdio7" name="B" time="0.216" file="wdio7">
      <error message="[2mexpect([22m[31mreceived[39m[2m).[22mtoBe[2m([22m[32mexpected[39m[2m) // Object.is equality[22m&#xA;&#xA;Expected: [32m&quot;B&quot;[39m&#xA;Received: [31m&quot;A&quot;[39m"/>
    </testcase>
  </testsuite>
</testsuites>
```

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### Reviewers: @webdriverio/project-committers
@BorisOsipov
@christian-bromann

### CODE REVIEW

File: packages/wdio-junit-reporter/src/index.ts [hunk 1]
```diff
@@ -173,6 +173,10 @@ class JunitReporter extends WDIOReporter {
                 testCase.skipped()
             } else if (test.state === 'failed') {
                 if (test.error) {
+                    if (test.error.message){
+                        test.error.message = test.error.message.replace(ansiRegex(), '')
```
Comment (christian-bromann) [inline packages/wdio-junit-reporter/src/index.ts:177]: I don't think we need to create a RegExp instance everytime here. We can just put what is being returned in `ansiRegex` at the top of the file.

**Review by christian-bromann — CHANGES_REQUESTED** [review]
Would be great if we can accompany this with a test.

File: packages/wdio-junit-reporter/src/index.ts (hunk 1 above)
Comment (unickq) [reply packages/wdio-junit-reporter/src/index.ts]: done

File: packages/wdio-junit-reporter/src/index.ts [hunk 2]
```diff
@@ -25,6 +25,11 @@ class JunitReporter extends WDIOReporter {
             : /[^a-zA-Z0-9@]+/ // Reason for ignoring @ is; reporters like wdio-report-portal will fetch the tags from testcase name given as @foo @bar
     }
 
+    private _ansiRegex = new RegExp([
```
Comment (christian-bromann) [inline packages/wdio-junit-reporter/src/index.ts:28]: No need to have it as class property either, I suggest to put it at the top

```js
const ansiRegex = new RegExp([...
```

**Review by christian-bromann — COMMENTED** [review]
One last comment

File: packages/wdio-junit-reporter/src/index.ts (hunk 2 above)
Comment (unickq) [reply packages/wdio-junit-reporter/src/index.ts]: ok

Comment (unickq) [reply packages/wdio-junit-reporter/src/index.ts]: It fails node 12 on the top 😩

**Review by christian-bromann — APPROVED** [review]
Awesome, thank you so much 👍

**Pull request merged**